TUCalendar/TUCalendar/Views/TableViewCells/TUCalendarWeekTableViewCell.m

```diff
@@ -74,16 +91,28 @@ - (void)layoutSubviews {
 
 - (void)setFirstDateOfWeek:(nonnull NSDate *)firstDayInWeek
             withDataSource:(nonnull id<TUCalendarWeekTableViewCellDataSource>)dataSource
-               monthNumber:(NSInteger)monthNumber
+          currentMonthDate:(nullable NSDate *)currentMonthDate
              departureDate:(nullable NSDate *)departureDate
                 returnDate:(nullable NSDate *)returnDate {
+    if (!self.weekCellAppearance) {
+        self.weekCellAppearance = [TUCalendarWeekTableViewCellAppearance new];
+    }
+
+    self.backgroundColor = self.weekCellAppearance.backgroundColor;
+    
     NSUInteger daysCount = self.daysViews.count;
 
     NSMutableArray *weekSettings = [NSMutableArray arrayWithCapacity:daysCount];
 
+    NSInteger monthNumber = [self.calendar component:NSCalendarUnitMonth fromDate:currentMonthDate];
+
     NSInteger departureDateMonth = [self.calendar component:NSCalendarUnitMonth fromDate:departureDate];
     NSInteger returnDateMonth = [self.calendar component:NSCalendarUnitMonth fromDate:returnDate];
 
+    NSInteger departureMonthAndYearFlag = departureDateMonth + [self.calendar component:kCFCalendarUnitYear fromDate:departureDate] * 12;
+    NSInteger returnMonthAndYearFlag = returnDateMonth + [self.calendar component:kCFCalendarUnitYear fromDate:returnDate] * 12;
+    NSInteger currentMonthAndYearFlag = monthNumber + [self.calendar component:kCFCalendarUnitYear fromDate:currentMonthDate] * 12;
```

Comment:
`12` - magic number. use constant or `rangeOfUnit:inUnit:forDate:` of `NSCalendar`